### PR TITLE
feat(#217) - Backend & DB for Home page - Creating basic dir structure and using the CLI generator and adding types

### LINKED ISSUE #217: Backend & DB for Home page - Creating basic dir structure and using the CLI generator and adding types
ONEPLAT-212
Currently, there is no way to store information displayed on the front end of the home SPA. 

## Describe the solution you'd like

Backend & DB for Home page

## Describe alternatives you've considered

None

### PR DESCRIPTION
**Pull request by hybridx**
feat(#217) - Created basic dir structure using CLI generator

Resolves #217 

# Explain the feature/fix

Basic dir structure using CLI generator and types for the home-spa

## Does this PR introduce a breaking change

No

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?

### CODE REVIEW

File: packages/home-service/src/resolver.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
-import { Home } from './schema';import { pubsub } from './helpers';
+import { Home } from './schema';
+import { pubsub } from './helpers';
 
-const Home_UPDATE = "Home_UPDATE";
+const Home_UPDATE = 'Home_UPDATE';
 
 export const HomeResolver = {
   Subscription: {
```
Comment (riginoommen) [inline packages/home-service/src/resolver.ts:7]: SUbscriptions don't work with federations directly. Please test your code

Comment (riginoommen) [inline packages/home-service/src/resolver.ts:7]: SUbscriptions don't work with federations directly. Please test your code

Comment (hybridx) [reply packages/home-service/src/resolver.ts:7]: That's not been worked upon in this PR, this is just the basic structure that has been generated by the CLI with types.

Comment (hybridx) [reply packages/home-service/src/resolver.ts]: @riginoommen I have removed the subscription as home-spa is not going to need it for now

File: packages/home-service/.env.example [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+GRAPHQL_API=https://localhost:8080/playground
+
+## Database
+DB_PATH=localhost
+DB_NAME=db_name
+# DB_USER=
+# DB_PASSWORD=
+
+## Certificate Configuration
+SSL_KEY=/etc/pki/tls/private/server.key
```
Comment (riginoommen) [inline packages/home-service/.env.example:18]:
```suggestion
# SSL_KEY=
```

File: packages/home-service/.env.example [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+SSL_CERT=/etc/pki/tls/certs/server.crt
```
Comment (riginoommen) [inline packages/home-service/.env.example:19]:
```suggestion
# SSL_CERT=
```

File: packages/home-service/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+Home Microservice
+=================================================
+
+One platform's server-side Home GraphQL microservice.
```
Comment (riginoommen) [inline packages/home-service/README.md:4]: Please add more information regarding your home spa

File: packages/home-service/service.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    const dbCredentials = (process.env.DB_USER && process.env.DB_PASSWORD)
+      ? `${process.env.DB_USER}:${process.env.DB_PASSWORD}@`
+      : '';
+    const dbConnection = `mongodb://${dbCredentials}${process.env.DB_PATH}/${process.env.DB_NAME}`;
+
+    mongoose.connect(dbConnection, { useNewUrlParser: true, useCreateIndex: true }).catch(console.error);
+
+    mongoose.connection.on('error', error => {
+      console.error(error);
+    });
```
Comment (riginoommen) [inline packages/home-service/service.ts:39]: Please change the alignment

Comment (hybridx) [reply packages/home-service/service.ts:39]: Alignment changed

File: packages/home-service/README.md (hunk 4 above)
Comment (hybridx) [reply packages/home-service/README.md:4]:
```
The Home SPA needs a backend API Service to fetch the microservice details and the Apps in spotlight list + videos. This microservice would solve that problem
by storing all such information. 
```
Added to readme

File: packages/home-service/.env.example (hunk 2 above)
Comment (hybridx) [reply packages/home-service/.env.example:18]: Was predefined, will change it

File: packages/home-service/.env.example (hunk 3 above)
Comment (hybridx) [reply packages/home-service/.env.example:19]: Was predefined, will change it

**Pull request merged**